File: Licensing/WinLicenseSDK/ExamplesSDK/SmartKeys Registration/VB .NET/ExampleSmartKeys/Form1.vb

Public Class Form1

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        Dim AppStatus As Short
        Dim Status As Short
        ' Check if the application is registered

        Status = WinlicenseSDK.WLRegGetStatus(AppStatus)

        ' if application registered, we display a messagebox and exit

        If Status = 1 Then
            MsgBox("The application is already REGISTERED", MsgBoxStyle.Information + MsgBoxStyle.OkOnly, "SmartKeys")
            Me.Close()
        End If

    End Sub

    Private Sub Label3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Label3.Click

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        Dim Status As Short

        Status = WinlicenseSDK.WLRegSmartKeyCheck(txtName.Text.ToString, txtCompanyName.Text.ToString,
                                                  txtCustomData.Text.ToString, txtActivationKey.Text.ToString)
        If Not Status Then

            MsgBox("Activation code is INVALID", MsgBoxStyle.Exclamation + MsgBoxStyle.OkOnly, "ERROR")

        Else

            ' activation data is correct. Now, we are going to install the smart key as File key 
            ' (via WLRegSmartKeyInstallToFile)
            WinlicenseSDK.WLRegSmartKeyInstallToFile(txtName.Text.ToString, txtCompanyName.Text.ToString,
                                                  txtCustomData.Text.ToString, txtActivationKey.Text.ToString)
            MsgBox("Activation code is correct. Product registered." & Chr(13) & Chr(10) &
                   "You must RESTART the application to finish the registration",
                   MsgBoxStyle.Information + MsgBoxStyle.OkOnly, "SmartKeys")

        End If

    End Sub
End Class

File: Licensing/WinLicenseSDK/ExamplesSDK/SmartKeys Registration/C#/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Runtime.InteropServices;
using System.IO;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();

             // Check if application is Registered
            int ExtendedInfo = 0;
            int ApplicationStatus = WinlicenseSDK.WLRegGetStatus(ref ExtendedInfo);

            if (ApplicationStatus == 1)
            {
                MessageBox.Show("The application is already REGISTERED", "SmartKeys");
                System.Environment.Exit(0);
            }
        }        
        private void ActivateButton_Click(object sender, EventArgs e)
        {
            bool IsSmartKeyCorrect = WinlicenseSDK.WLRegSmartKeyCheck(
                                            this.NameEdit.Text, this.CompanyEdit.Text,
                                            this.CustomEdit.Text, this.SmartKeyEdit.Text);
            if (!IsSmartKeyCorrect)
            {
                MessageBox.Show("Activation code is INVALID", "ERROR");
            }
            else
            {
                // activation data is correct. Now, we are going to install the smart key as File key 
                // (via WLRegSmartKeyInstallToFile)
                WinlicenseSDK.WLRegSmartKeyInstallToFile(
                        this.NameEdit.Text, this.CompanyEdit.Text,
                        this.CustomEdit.Text, this.SmartKeyEdit.Text);
                MessageBox.Show("Activation code is CORRECT. Product Registered\nYou must RESTART" +
                                "he application to finish the registration", "SmartKeys");
            }
        }
    }    
}